src/dynamics_tendencies.jl

```diff
@@ -0,0 +1,216 @@
+    end
+
+    for k in 1:nlev
+        u_tend[:,:,k] = u_tend[:,:,k] 
+                        + v_grid[:,:,k].*vor_grid[:,:,k] 
+                        - temp_grid_anomaly[:,:,k]*px
+                         - (arbitrary_array[:,:,k+1] + arbitrary_array[:,:,k])*dhsr[k]
+    end
+
+end
+
+
+
+"""
+Compute the spectral tendency of the meridional wind 
+"""
+function meridional_wind_tendency!(Diag::PrognosticVariables{NF}, # Diagnostic variables
+                                   C::Constants{NF}
+                                  )where {NF<:AbstractFloat}
+
+    @unpack v_tend = Diag.Tendencies
+    @unpack vor_grid,u_grid,v_grid,temp_grid_anomaly =Diag.gridvars
+    @unpack sigma_tend,arbitrary_array,px,py = Diag.miscvars
+    @unpack nlev,dhsr= C 
+
+
+    for k in 2:nlev
+        arbitrary_array[:,:,k] = sigma_tend[:,:,k].*(v_grid[:,:,k] - v_grid[:,:,k-1])
+    end
+                            
+    for k in 1:nlev
+        v_tend[:,:,k] = v_tend[:,:,k] 
+                        -u_grid[:,:,k].*vor_grid[:,:,k] 
+                        - temp_grid_anomaly[:,:,k]*py
+                        - (arbitrary_array[:,:,k+1] + arbitrary_array[:,:,k])*dhsr[k]
+    end
+
+end
+
+
+
+
+
+"""
+Compute the spectral temperature tendency
+"""
+function temperature_tendency!(Diag::PrognosticVariables{NF}, # Diagnostic variables
+                               C::Constants{NF}
+                               )where {NF<:AbstractFloat}
+    
+    @unpack temp_tend = Diag.Tendencies
+    @unpack div_grid,temp_grid_anomaly =Diag.gridvars
+    @unpack d_mean,arbitrary_array,sigma_tend,sigma_m,puv= Diag.miscvars
+    @unpack tref,tref3,nlev,dhsr,fsgr,akap = C
+
+
+
+
+    for k in 2:nlev
+        arbitrary_array[:,:,k] = sigma_tend[:,:,k].*(temp_grid_anomaly[:,:,k] - temp_grid_anomaly[:,:,k-1])
+                    + sigma_m[:,:,k].*(tref[k] - tref[k-1])
+    end
+
+    for k in 1:nlev
+        temp_tend[:,:,k] = temp_tend[:,:,k]
+                        + temp_grid_anomaly[:,:,k].*div_grid[:,:,k]
+                        - (arbitrary_array[:,:,k+1] + arbitrary_array[:,:,k])*dhsr[k]
+                        + fsgr[k]*temp_grid_anomaly[:,:,k].*(sigma_tend[:,:,k+1] + sigma_tend[:,:,k])
+                        + tref3[k]*(sigma_m[:,:,k+1] + sigma_m[:,:,k])
+                        + akap*(t_grid[:,:,k].*puv[:,:,k] - temp_grid_anomaly[:,:,k].*d_mean)
+    end 
+
+
+end
+
+
+
+
+"""
+Compute the humidity tendency
+"""
+function humidity_tendency!(Diag::PrognosticVariables{NF}, # Diagnostic variables
+                          C::Constants{NF}
+                          )where {NF<:AbstractFloat}
+    
+
+    @unpack div_grid,humid_grid = Diag.gridvars
+    @unpack arbitrary_array,sigma_tend= Diag.miscvars
+    @unpack dhsr, nlev = C
+
+    for k in 2:nlev
+        arbitrary_array[:,:,k] = sigma_tend[:,:,k].*(humid_grid[:,:,k,itr] - humid_grid[:,:,k-1,itr])
+    end
+    
+    arbitrary_array[:,:,2:3] .= zero
```

Comment:
`zero` is a function that requires as a first argument a type. E.g. `zero(NF)`

Reply:
Fixed